htdocs/core/db/mssql.class.php

```diff
@@ -324,9 +325,13 @@ function rollback($log='')
 	 */
 	function query($query,$usesavepoint=0,$type='auto')
 	{
-		$query = trim($query);
+        global $debugBar;
 
-		if (preg_match('/^--/',$query)) return true;
+        $query = trim($query);
+        if (isset($debugBar))
```

Comment:
You should not need to add code into db driver. Into function query, at end, there is already a call to 
dol_syslog
And this function include an architecture to allow to add more logging output (you should fill the ata you want to report into a log driver (you may have an examples into
core/modules/syslog
Each file mod_syslog_.... is a driver to manage a different output.